### ISSUE TITLE
Stacks On Chain Portfolio

### ISSUE THREAD

**Issue by cbadawi** [stacksgov/Stacks-Grant-Launchpad#269]
**Background**
This is a followup to the previous grant #203 , keeping course with DeFi and NFT analytics.

**Project Overview**
- Currently there is no way to keep track of all your investments in STX or USD terms. We'll offer a solution for that. Taking inspiration from [Zerion](https://app.zerion.io/0x5b3ce67ebc795fe7e709815bc49d4300898e1b7b/overview) and [Nansen.ai](https://www.nansen.ai/resources/introducing-wallet-profiler) 's wallet profiler.

-  As a continuation to our previous work creating the floor prices queries, NFT stats, token prices... : Stacks on Chain will offer the product for Free to the community helping everyone to keep track of all their investments in one place, whether its an NFT, Diko, Alex and other tokens or STX. We will also  enrich it with more features not present on Zerion nor Nansen. The long term objective is for  this project to transition to a portfolio management system where users will be able to balance pools or list/unlist their NFTs, and interact with other smart contracts, but thats for a later date.

**Budget and Milestones**
- M1: Stacks on Chain Portfolio : $16,000
By the end of this milestone, we will deliver  a complete product detailing your NFT, tokens and STX in your wallets. We expect M1 to take 3 weeks.

- M2: *Bonus* Stacks on Chain Dashboard Competition
 $4,000
As an extension to our first application  #203 we would also like to start a competition on Stacks on Chain. Users who created the most viewed (or most voted on) dashboards will receive NFTs airdropped to their wallet. We think this is a great way for the community to get together, learn and build. 
Since a good dashboard would take a weekend to be built, maybe longer for beginners, we think the prize should be worth the effort users will put into building the dashboards. Thus, we are asking for $4000 to challenge users to learn and giveaway a megapont ape and another blue chip NFT at a lower price for second place.
I am really open for ideas about this and who to partner with to increase visibility to the entire ecosystem.
The competition can only start after prior M1 is completed as we will be spending significant amount of time creating education content, resources and helping people write their own queries. We will not take any cut from this and will try to spend 100% of the grant on the prizes.
We expect the competition to last 4 to 6 weeks, that is enough time for users to get acquainted with the data and produce quality dashboards.

Total Grant Request:  $20,000

M1: $16,000
We expect the Portfolio project to take 3 weeks to complete.

M2: $4,000
We expect the Competition to take place over a period of 6 weeks.

**Risks**
There are 0 risks.

Community and Supporting Materials
My previous work: #203.

**Comment by tycho1212** [comment]
I think codex.btc was working on something similar as a pet project. You can probably ask him to share some of his code. He's on twitter :)

**Comment by cbadawi** [comment]
@tycho1212 yep thanks. Out of courtesy I already did ask him and offered him our api, but he has not been working on it lately.

**Comment by friedger** [comment]
Please make it so that I can see the protofolio including more than my own address (but all of my other accounts as well - as read-only)

**Comment by cbadawi** [comment]
@friedger for sure. it is one of the most common requests.

**Comment by will-at-stacks** [comment]
Hi @cbadawi - Thank you for the proposal.  Sounds like a great project.  We are approving this proposal, but would like you to provide a mock-up of the on chain portfolio early in the process.  Once that is ready to go, please share it with us and the community (however you see fit) so we can make sure feedback is heard.  Our hope is not to micro-manage your creativity, but rather help this be the most useful tool it can be.  Lastly, we have some great community engagement people that could help shape the dashboard competition component.  Would you be open to meeting with them so we get the most engagement from this process?  Lets connect when you are ready to brainstorm this part of the project.  Best, Will

**Comment by cbadawi** [comment]
awesome thanks @will-at-stacks . will share a mock-up for feedback within a couple of days.

**Comment by cbadawi** [comment]
hey @will-at-stacks , shared the initial mock up here
https://twitter.com/anononchain/status/1488549598281240576

**Comment by cbadawi** [comment]
hey @will-at-stacks ! released the portfolio here. let me know if you have any remarks. https://twitter.com/anononchain/status/1500882683165315074

**Comment by cbadawi** [comment]
!m2_complete

**Comment by will-at-stacks** [comment]
Hello and thank you for participating in the Stacks Foundation Grants Program!

We are in the process of migrating from GitHub to the new [Grants Dashboard](https://grants.stacks.org/). In order to complete your grant, you will need to submit any remaining Progress Review and/or Final Review requests through the Dashboard in order to receive your remaining payments.

- If you have questions about the Grants Dashboard in general please email [ambassadors@stacks.org](mailto:ambassadors@stacks.org).
- If you have questions regarding payments and/or contracts please email [payments@stacks.org](mailto:payments@stacks.org).

Lastly, please note we are marking this grant 'closed' on GitHub for organizational purposes, but it is still 'open' on the Grants Dashboard.

Thanks and we hope to continue to support your efforts with additional grants!

Best,
Will
